### PR TITLE
Add 3dsecure.io integration (GWY-15016)

### PR DESCRIPTION
**Pull request by agalstian**
### Details

This pr is adding fields that are needed for integrating 3dsecure.io.

### Links
https://github.com/Rebilly/backend/pull/7877
https://jira.ilabsinc.com/browse/GWY-15016

### CODE REVIEW

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+description: ThreeDSecureIOMpi
+allOf:
+  - type: object
```
Comment (edmandiesamonte) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]: Not sure about this but I noticed that in most MPI schemas, the actual definition are in separate file. I don't know the exact reason about it, but would it be better to stay consistent with it?

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    required:
+      - merchantId
+      - merchantName
+      - merchantAcquirerBINVisa
```
Comment (adamaltman) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]:
```suggestion
      - merchantAcquirerBinVisa
```
camelCase

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      - merchantAcquirerBINMastercard
```
Comment (adamaltman) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]:
```suggestion
      - merchantAcquirerBinMastercard
```

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      merchantId:
+        type: string
+        description: Merchant Id
+      merchantName:
+        type: string
+        format: password
+        description: Merchant Name. Maximum length should be 25 characters.
+      merchantAcquirerBINVisa:
+        type: string
+        description: Merchant Visa Acquired BIN
```
Comment (adamaltman) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]:
```suggestion
        description: Visa Acquirer BIN
```

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      - merchantUrl
+    properties:
+      merchantId:
+        type: string
+        description: Merchant Id
+      merchantName:
+        type: string
+        format: password
+        description: Merchant Name. Maximum length should be 25 characters.
+      merchantAcquirerBINVisa:
```
Comment (adamaltman) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]:
```suggestion
      merchantAcquirerBinVisa:
```

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        type: string
+        description: Merchant Visa Acquired BIN
+      merchantAcquirerBINMastercard:
```
Comment (adamaltman) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]:
```suggestion
      merchantAcquirerBinMastercard:
```

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+          type: string
+          description: Merchant Mastercard Acquired BIN
```
Comment (adamaltman) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]:
```suggestion
          description: Mastercard Acquirer BIN
```

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml (hunk 1 above)
Comment (adamaltman) [reply openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]: No reason to do that.

**Comment by adamaltman** [comment]
I don't see this as usable by any gateway in the API definition.

**Comment by agalstian** [comment]
@adamaltman right, it's not used by gateways. as far as i understood connecting it to a gateway would be another ticket.

**Comment by slavcodev** [comment]
@agalstian jfyi, github also supports applying suggestions in batch with one commit.

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+      - merchantCountry
+      - merchantUrl
+    properties:
+      merchantId:
+        type: string
+        description: Merchant Id
+      merchantName:
+        type: string
+        format: password
+        description: Merchant Name. Maximum length should be 25 characters.
```
Comment (ashkarpetin) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]:
```suggestion
        maxLength: 25
        description: Merchant Name
```

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+      merchantAcquirerBinVisa:
+        type: string
+        description: Visa Acquirer BIN
+      merchantAcquirerBinMastercard:
+          type: string
+          description: Mastercard Acquirer BIN
+      merchantCountry:
+        type: string
+        description: Merchant Country Code. Should be 2 characters long, e.g. US, GB.
```
Comment (ashkarpetin) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]:
```suggestion
        maxLength: 2
        description: Merchant Country ISO Alpha-2 Code. 
        example: US
```

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+      - merchantAcquirerBinMastercard
+      - merchantCountry
+      - merchantUrl
+    properties:
+      merchantId:
+        type: string
+        description: Merchant Id
+      merchantName:
+        type: string
+        format: password
```
Comment (ashkarpetin) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]: Why `merchantName` is masked?

File: openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        description: Visa Acquirer BIN
+      merchantAcquirerBinMastercard:
+          type: string
+          description: Mastercard Acquirer BIN
+      merchantCountry:
+        type: string
+        description: Merchant Country Code. Should be 2 characters long, e.g. US, GB.
+      merchantUrl:
+        type: string
+        description: Merchant URL
```
Comment (ashkarpetin) [inline openapi/components/schemas/Mpi/ThreeDSecureIOMpi.yaml]:
```suggestion
        description: Merchant URL.
```

**Pull request merged**